File: app/src/main/java/com/example/pictureme/views/friends/FriendRequestsFragment.java
package com.example.pictureme.views.friends;

import android.os.Bundle;

import androidx.fragment.app.Fragment;
import androidx.recyclerview.widget.LinearLayoutManager;
import androidx.recyclerview.widget.RecyclerView;

import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;

import com.example.pictureme.R;
import com.example.pictureme.views.adapters.FriendListAdapter;
import com.example.pictureme.views.adapters.FriendRequestsAdapter;

public class FriendRequestsFragment extends Fragment {

    int dummyFriendsImages[] = {R.drawable.face, R.drawable.face};

    RecyclerView recyclerView;

    @Override
    public View onCreateView(LayoutInflater inflater, ViewGroup container,
                             Bundle savedInstanceState) {
        // Inflate the layout for this fragment
        View view = inflater.inflate(R.layout.fragment_friend_requests, container, false);

        recyclerView = view.findViewById(R.id.recycler_view_requests);

        String[] friendNames = getResources().getStringArray(R.array.dummy_friends);

        FriendRequestsAdapter myAdapter = new FriendRequestsAdapter(view.getContext(), friendNames, dummyFriendsImages);
        recyclerView.setAdapter(myAdapter);
        recyclerView.setLayoutManager(new LinearLayoutManager(view.getContext()));

        return view;
    }
}

File: app/src/main/java/com/example/pictureme/views/profile/friends/tabs/friendRequests/FriendRequestsFragment.kt
package com.example.pictureme.views.profile.friends.tabs.friendRequests

import android.os.Bundle
import androidx.fragment.app.Fragment
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import androidx.fragment.app.activityViewModels
import androidx.recyclerview.widget.LinearLayoutManager
import com.example.pictureme.data.models.FriendRequest
import com.example.pictureme.databinding.FragmentFriendRequestsBinding
import com.example.pictureme.viewmodels.UserViewModel
import com.example.pictureme.views.profile.friends.tabs.friendsList.FriendsListAdapter
import dagger.hilt.android.AndroidEntryPoint

@AndroidEntryPoint
class FriendRequestsFragment : Fragment() {
    private var _binding: FragmentFriendRequestsBinding? = null
    private val binding get() = _binding!!

    private val userViewModel by activityViewModels<UserViewModel>()

    private var friendRequests: List<FriendRequest> = emptyList()

    override fun onCreateView(
        inflater: LayoutInflater, container: ViewGroup?,
        savedInstanceState: Bundle?
    ): View? {
        _binding = FragmentFriendRequestsBinding.inflate(inflater, container, false)

        userViewModel.loadUser()

        setFriendRequestsAdapter()

        return (binding.root)
    }

    private fun setFriendRequestsAdapter() {
        val adapter = FriendRequestsAdapter(emptyList(), userViewModel)
        binding.requestsRecyclerView.adapter = adapter
        binding.requestsRecyclerView.layoutManager = LinearLayoutManager(activity)

        userViewModel.userLiveData.observe(viewLifecycleOwner) { user ->
            println("CHANGING REQUEST LIST")
            friendRequests = user.friendRequests
            println(friendRequests)
            adapter.setFriendRequests(friendRequests)
        }
    }

    override fun onDestroyView() {
        super.onDestroyView()
        _binding = null
    }
}